File: src/lib/library/v1/copy.ts
import { GpuDesignerNode } from "../../designer/gpudesignernode";

export class CopyNode extends GpuDesignerNode {
	public init() {
		this.title = "Copy";

		this.addInput("image");
		this.addStringProperty("name", "Name");

		const source = `
        vec4 process(vec2 uv)
        {
            vec4 col = texture(image, uv);
            return col;
        }
        `;

		this.buildShader(source);
	}
}

File: src/texturelab/libraries/v1/copy.cpp
#include "../../models.h"
#include "../../props.h"
#include "../libv1.h"

void CopyNode::init()
{
    this->title = "Copy";

    this->addInput("image");
    this->addStringProp("name", "Name");

    auto source = R""""(
        vec4 process(vec2 uv)
        {
            vec4 col = texture(image, uv);
            return col;
        }
        )"""";

    this->setShaderSource(source);
}